node/node_version_check.go

```diff
@@ -0,0 +1,32 @@
+package node
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spacemeshos/go-spacemesh/config"
+	"github.com/spacemeshos/go-spacemesh/sql"
+)
+
+func (app *App) verifyVersionUpgrades() error {
+	return verifyUpgradeToV15(app.Config)
+}
+
+// v1.5 requires going through v1.4 first as it removed in-code migrations.
+func verifyUpgradeToV15(cfg *config.Config) error {
```

Comment:
This can probably be inlined, because if we require a higher version later we will just upgrade the version check on line 28 below instead of adding a new function.